e2e/cypress/e2e/loginFlow.cy.js

```diff
@@ -0,0 +1,128 @@
+const nextjsBaseUrl = Cypress.env("nextjsBaseUrl");
+
+describe(`Different ways to open login modal with library with borrowerCheck `, () => {
+  it("Plain login from login button in header", () => {
+    cy.visit(nextjsBaseUrl);
+    cy.consentAllowAll(); //allow cookies
+    cy.get("[data-cy=header-link-login]").should("be.visible").click();
+    cy.get("[data-cy=pickup-search-input]").should("be.visible").type("Val");
+    cy.contains("Valby Bibliotek").click();
+    cy.get("[data-cy=button-log-ind]").should("be.visible");
+  });
+
+  it("Plain login from login button on profile page", () => {
+    cy.visit(nextjsBaseUrl + "/profil");
+    cy.consentAllowAll();
+    cy.get('[class^="ProfileLayout_loginButton__sdKEd"]') //we can have two login buttons on the page, take the one where classname starts with this
+      .should("be.visible")
+      .click();
+    cy.get("[data-cy=text-log-ind]").should("be.visible");
+  });
+
+  it("Infomedia shows correct login modal", () => {
+    cy.visit(
+      nextjsBaseUrl +
+        "/infomedia/politiken/work-of:870971-avis:87043134/Y4074532" //TODO could this article disappear? should i mock this here?
+    );
+    cy.consentAllowAll();
+    cy.get("[data-cy=button-log-ind]")
+      .filter((index, element) => {
+        const classes = element.className.split(" ");
+        return classes.some((cls) => cls.includes("primary")); //we can have two login buttons on the page, where class contains "primary"
+      })
```

Comment:
Du kan ikke injecte en dataCy til de dersens knapper i stedet for at gøre det her?